mmv1/third_party/terraform/tests/resource_redis_cluster_test.go

```diff
@@ -0,0 +1,108 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+// Validate that replica count is updated for the cluster
+func TestAccRedisCluster_updateReplicaCount(t *testing.T) {
```

Comment:
Acceptance test are failing with the following error and I was unable to figure out why. 

=== NAME  TestAccRedisCluster_updateReplicaCount
    vcr_utils.go:152: Step 1/5 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        An execution plan has been generated and is shown below.
        Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # google_redis_cluster.test will be updated in-place
          ~ resource "google_redis_cluster" "test" {
                create_time             = "2023-08-04T21:48:16.655860311Z"
                discovery_endpoints     = [
                    {
                        address    = "10.128.15.236"
                        port       = 6379
                        psc_config = [
                            {
                                network = "projects/1044042403698/global/networks/default"
                            },
                        ]
                    },
                ]
                id                      = "projects/khimani-redis-test/locations/us-central1/clusters/tf-test-2677204341121549155"
                name                    = "tf-test-2677204341121549155"
                project                 = "khimani-redis-test"
                region                  = "us-central1"
                replica_count           = 1
                shard_count             = 3
                size_gb                 = 39
                state                   = "ACTIVE"
                transit_encryption_mode = "TRANSIT_ENCRYPTION_MODE_DISABLED"
                uid                     = "92d0168c-9294-4cf8-97a6-afb8f1d914ce"
        
              + psc_configs {
                  + network = "projects/1044042403698/global/networks/default"
                }
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
    testing_new.go:84: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: Instance cannot be destroyed
        
          on terraform_plugin_test.tf line 2:
           2: resource "google_redis_cluster" "test" {
        
        Resource google_redis_cluster.test has lifecycle.prevent_destroy set, but the
        plan calls for this resource to be destroyed. To avoid this error and continue
        with the plan, either disable lifecycle.prevent_destroy or reduce the scope of
        the plan using the -target flag.

Reply:
This looks like the field `psc_configs` is either not being returned by the API or not being read correctly by Terraform. Checking the debug logs may help you figure out which one is going on

Reply:
http://yaqs/7048917368625430528 The output doesn't have psc_configs field, rather it is nested within discovery_endpoints as you can see from the output above. What would be the best way to handle this scenario?